examples/routing/src/lib.rs

```diff
@@ -32,40 +34,21 @@ impl Component for Model {
 
     fn create(_: Self::Properties, mut link: ComponentLink<Self>) -> Self {
         let callback = link.send_back(|route: Route<()>| Msg::HandleRoute(route));
-        let mut router = router::Router::bridge(callback);
-
-        // TODO Not sure if this is technically correct. This should be sent _after_ the component has been created.
-        // I think the `Component` trait should have a hook called `on_mount()`
-        // that is called after the component has been attached to the vdom.
-        // It seems like this only works because the JS engine decides to activate the
-        // router worker logic after the mounting has finished.
-        router.send(router::Request::GetCurrentRoute);
-
+        let router = router::Router::bridge(callback);
         Model {
-            child: Child::A, // This should be quickly overwritten by the actual route.
+            child: Child::Loading, // This should be quickly overwritten by the actual route.
             router,
         }
     }
 
+    fn mounted(&mut self) -> ShouldRender {
+        // Once the component has successfully mounted, it is safe to send messages to agents.
```

Comment:
I'm not sure if this is technically correct. I think it's only safe to send messages when the agent notifies the handler that it is connected currently. Although that will change after this is merged: https://github.com/yewstack/yew/pull/596

Reply:
Good point, I'll remove the comment. I do think its a good practice to send messages in `mounted`, and leave `create` for creating stuff only.